src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -729,7 +729,7 @@ InterpreterCreateQuery::TableProperties InterpreterCreateQuery::getTableProperti
     }
     /// We can have queries like "CREATE TABLE <table> ENGINE=<engine>" if <engine>
     /// supports schema inference (will determine table structure in it's constructor).
-    else if (!StorageFactory::instance().checkIfStorageSupportsSchemaInterface(create.storage->engine->name)) // NOLINT
+    else if (create.storage && create.storage->engine && !StorageFactory::instance().checkIfStorageSupportsSchemaInterface(create.storage->engine->name)) // NOLINT
```

Comment:
Then you need to remove `// NOLINT` - this is an annotation for clang-tidy (another static analyzer).

Reply:
Also, need to throw a logical error if the engine is nullptr.
Otherwise, you assume it can be nullptr, but you are leaving the code running in an inconsistent state.

Reply:
I appreciate the comments. I'll make sure to incorporate these changes.